### PR TITLE
build tools and sdk update

### PR DESCRIPTION
**Pull request by jasonalmaturner**
Update the Android build tools and sdk to work with React Native 0.56.

### CODE REVIEW

**Comment by sadafk831** [comment]
@jasonalmaturner Thank you for the changes, would you mind making the build tools use the root project versions like following:
```
def safeExtGet(prop, fallback) {
    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
}

compileSdkVersion safeExtGet('compileSdkVersion', 26)
```

**Comment by jasonalmaturner** [comment]
Totally. That's a much better solution.

File: android/build.gradle [hunk 1]
```diff
@@ -5,19 +5,23 @@ buildscript {
...
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.3"
+    compileSdkVersion safeExtGet('compileSdkVersion', 26)
+    buildToolsVersion "26.0.3"
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 23
+        targetSdkVersion 26
```
Comment (jasonalmaturner) [inline android/build.gradle]: Should I use `safeExtGet` for the `targetSdkVersion` as well?

Comment (skb1129) [reply android/build.gradle]: Yes, I have seen most of the react-native packages using this on `buildToolsVersion` and `targetSdkVersion` as well.

**Comment by skb1129** [comment]
This is required to be released soon to work with RN 0.56

File: android/build.gradle [hunk 2]
```diff
@@ -17,11 +17,11 @@ def safeExtGet(prop, fallback) {
 
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', 26)
-    buildToolsVersion "26.0.3"
+    buildToolsVersion safeExtGet('compileSdkVersion', "26.0.3")
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 26
+        targetSdkVersion safeExtGet('compileSdkVersion', 26)
```
Comment (daku) [inline android/build.gradle]: Should this be `safeExtGet('targetSdkVersion', 26)`?

File: android/build.gradle [hunk 3]
```diff
@@ -5,19 +5,23 @@ buildscript {
...
 
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.3"
+    compileSdkVersion safeExtGet('compileSdkVersion', 26)
+    buildToolsVersion safeExtGet('compileSdkVersion', "26.0.3")
```
Comment (daku) [inline android/build.gradle]: This one needs to be `safeExtGet('buildToolsVersion', "26.0.3")`

Comment (jasonalmaturner) [reply android/build.gradle]: Gosh. I'm dumb. Yes. I'll fix that right now.

**Comment by p1zzadog** [comment]
What's the timeline for this being merged and released? We need this for an upcoming release and I'd like to avoid forking this project.

**Comment by 4ian** [comment]
@tychota @Minishlink @Almouro @AuroreM @chdeps @yleflour Anyone at BAM that could double check this PR and release it? I'm not working on a RN app right now so I have little time for this.

**Comment by Minishlink** [comment]
Will try to look into this.

In the meantime you can still use the following in your root `build.gradle`

```groovy
subprojects { subproject ->
    afterEvaluate{
        if((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion
            }
        }
    }
}
```

**Pull request merged**